keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/Hamiltonian.java

```diff
@@ -22,36 +26,53 @@
  * The No-U-Turn Sampler: Adaptively Setting Path Lengths in Hamiltonian Monte Carlo
  * https://arxiv.org/pdf/1111.4246.pdf
  */
-public class Hamiltonian {
+@Builder
+public class Hamiltonian implements PosteriorSamplingAlgorithm {
 
-    private Hamiltonian() {
+    private static final double DEFAULT_STEP_SIZE = 0.1;
+    private static final int DEFAULT_LEAP_FROG_COUNT = 20;
+
+    public static Hamiltonian withDefaultConfig() {
+        return withDefaultConfig(KeanuRandom.getDefaultRandom());
+    }
+
+    public static Hamiltonian withDefaultConfig(KeanuRandom random) {
+        return Hamiltonian.builder()
+            .random(random)
+            .leapFrogCount(DEFAULT_LEAP_FROG_COUNT)
+            .stepSize(DEFAULT_STEP_SIZE)
```

Comment:
Given we have a default value specified for these - do we need to specify them as part of the builder here?  Seems redundant (repeated comment as my previous one seems to have disappeared for some reason)

Reply:
Just for clarity if someone looks a the decompiled class. I've moved them to static finals so at least it's ensured they're the same.